# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/public/experimental/TypedData.h
# Commit: 16695064a637
# Full Hash: 16695064a637776379f4cb777771087cf12c0760
# Author: Steve Fink <sfink@mozilla.com>
# Date: 2021-08-13 09:27:46
# Regressor Bug: 1720422
# File Overlap Count: 2
# Description:
#   Bug 1720422 - Convert the JS_Is(type)Array APIs r=jonco
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D119860
# ==============================================================================

diff -r b5bafd6f08d3 -r 16695064a637 js/public/experimental/TypedData.h
--- a/js/public/experimental/TypedData.h	Thu Aug 12 20:44:17 2021 +0000
+++ b/js/public/experimental/TypedData.h	Thu Aug 12 20:44:17 2021 +0000
@@ -114,20 +114,6 @@
  */
 extern JS_PUBLIC_API bool JS_IsArrayBufferViewObject(JSObject* obj);
 
-/*
- * Test for specific typed array types (ArrayBufferView subtypes)
- */
-
-extern JS_PUBLIC_API bool JS_IsInt8Array(JSObject* obj);
-extern JS_PUBLIC_API bool JS_IsUint8Array(JSObject* obj);
-extern JS_PUBLIC_API bool JS_IsUint8ClampedArray(JSObject* obj);
-extern JS_PUBLIC_API bool JS_IsInt16Array(JSObject* obj);
-extern JS_PUBLIC_API bool JS_IsUint16Array(JSObject* obj);
-extern JS_PUBLIC_API bool JS_IsInt32Array(JSObject* obj);
-extern JS_PUBLIC_API bool JS_IsUint32Array(JSObject* obj);
-extern JS_PUBLIC_API bool JS_IsFloat32Array(JSObject* obj);
-extern JS_PUBLIC_API bool JS_IsFloat64Array(JSObject* obj);
-
 /**
  * Return the isShared flag of a typed array, which denotes whether
  * the underlying buffer is a SharedArrayBuffer.
@@ -617,4 +603,17 @@
 
 }  // namespace JS
 
+/*
+ * JS_Is(type)Array(JSObject* maybeWrapped)
+ *
+ * Test for specific typed array types.
+ */
+
+#define DECLARE_IS_ARRAY_TEST(_1, _2, Name)                                   \
+  inline JS_PUBLIC_API bool JS_Is##Name##Array(JSObject* maybeWrapped) {      \
+    return JS::TypedArray<js::Scalar::Name>::unwrap(maybeWrapped).asObject(); \
+  }
+JS_FOR_EACH_TYPED_ARRAY(DECLARE_IS_ARRAY_TEST)
+#undef DECLARE_IS_ARRAY_TEST
+
 #endif  // js_experimental_TypedData_h